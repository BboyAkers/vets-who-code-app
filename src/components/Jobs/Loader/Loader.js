import PropTypes from 'prop-types'
import './loader.css'

function Loader({ isSubmitted, jobData }) {
  return (
    <div className={`loading ${isSubmitted && jobData === false ? '' : 'hidden'}`}>
      <div className="ball first"></div>
      <div className="ball second"></div>
      <div className="ball third"></div>
    </div>
  )
}

export default Loader

Loader.propTypes = {
  isSubmitted: PropTypes.bool,
  jobData: PropTypes.oneOfType([ 
      PropTypes.bool,
      PropTypes.object
  ])
}
