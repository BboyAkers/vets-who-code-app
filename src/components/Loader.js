import React, { Component } from 'react';

class Loader extends Component {
  componentDidMount() {
    global.jQuery = require('../../static/vendor/jquery/dist/jquery');
    window.$ = jQuery;
    window.jQuery = jQuery;
    this.handleLoader();
  }

  handleLoader = () => {
    $(window).on('load', () => {
      $('#loader').fadeOut();
      $('#loader-wrapper')
        .delay(100)
        .fadeOut();
    });
  }

  render() {
    return (
      <div id="loader-wrapper">
        <div id="loader" />
      </div>
    );
  }
}

export default Loader;
