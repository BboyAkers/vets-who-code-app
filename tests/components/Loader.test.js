import React from 'react'
import Loader from '../../src/components/Loader'

describe('<Loader />', () => {
  let wrapper

  beforeEach(() => wrapper = shallow(<Loader />))
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
