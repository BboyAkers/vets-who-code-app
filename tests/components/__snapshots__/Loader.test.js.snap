// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly 1`] = `
<div
  id="loader-wrapper"
>
  <div
    id="loader"
  />
</div>
`;
